### DISCUSSION TITLE
Goneovim cannot display `cmdline_popup`

### DISCUSSION THREAD

**Discussion by przepompownia** [folke/noice.nvim#212] [General]
I use master version of Neovim, built from sources everyday. After numerous bugs with redrawing from some moment (on Wayland) I switched from nvim-qt to goneovim (https://github.com/akiyosi/goneovim). 

Looking into https://github.com/rcarriga/nvim-notify/issues/5 and https://github.com/neovim/neovim/pull/16396 I decided to check what it is Noice. Unfortunately, `cmdline_popup` turned out not to work in Goneovim. In terminal (kitty) command line popup works. 

If you can reproduce it, then probably we need to warn users similarly to Neovide (docs + checkhealth).

**Comment by przepompownia** [comment]
Probably the problem is not as general as I described it above: I can display nui input:
```lua
require('nui.input')({position = '50%', size = {width = 50}}, {prompt = 'Goneovim > ', default_value = 'works'}):mount()
```
![image](https://user-images.githubusercontent.com/11404453/201347073-b01216e1-61d3-4ea8-8d19-442d8b11d782.png)
Please note that the command still displayed in nvim's built-in cmdline. The same in terminal:

https://user-images.githubusercontent.com/11404453/201349886-5a510018-0b1a-4917-87dc-21a4c25ee76f.mp4

**Comment by przepompownia** [comment]
I can also reproduce it with the minimal config `/tmp/noice-min.lua` from https://github.com/folke/noice.nvim/wiki/Minimal-%60init.lua%60-to-Reproduce-an-Issue

```sh
goneovim --nvim ~/tmp/nvim-from-src/bin/nvim -u /tmp/noice-min.lua
```

**Reply by folke** [reply]
Most likely a Goneovim problem. Especially since it works in Tui.

**Reply by folke** [reply]
Goneovim externalizes `multigrid`. With that ext enabled, Noice cant work

**Reply by folke** [reply]
I've added additional health checks for multigrid until that is fixed on the Neovim side.

**Reply by przepompownia** [reply]
@folke thanks for adding warning mesages and checkhealth, especially to avoid confusing new users. I never had not need to read `ui.txt` but maybe it's the time. 

It's a chance that [I can switch back to neovim-qt](https://github.com/equalsraf/neovim-qt/issues/1028) on which Noice displays `cmdline_popup` correctly.

**Reply by folke** [reply]
In theory it should work, but something currently prevents messages and cmldine events when multigrid is enabled. See https://github.com/neovim/neovim/issues/21068

**Comment by folke** [comment]
I have a [Neovim PR](https://github.com/neovim/neovim/pull/21080) that fixes the problems with multigrid, but Goneovim will also need to make a change.

See here https://github.com/akiyosi/goneovim/issues/423
